packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -286,6 +288,30 @@ export async function handleRoute({
 	// By default, we should give priority to the status code passed, although it's possible that
 	// the `Response` emitted by the user is a redirect. If so, then return the returned response.
 	if (response.status < 400 && response.status >= 300) {
+		if (
+			response.status >= 300 &&
+			response.status < 400 &&
+			routeIsRedirect(route) &&
+			!config.build.redirects &&
+			pipeline.settings.buildOutput === 'static'
+		) {
+			// If we're here, it means that the calling static redirect that was configured by the user
+			// We try to replicate the same behaviour that we provide during a static build
+			response = new Response(
+				redirectTemplate({
+					delay: response.status === 302 ? 2 : 0,
```

Comment:
Is there something special about the 2 value?

Reply:
I refactored the code to keep the computation of `delay` inside one place. This was already documented in the `generate.ts` file: https://developers.google.com/search/docs/crawling-indexing/301-redirects#metarefresh

Essentially, for static files, to simulate a `302` redirect - which are *temporary* -, we need an arbitrary delay so Google understand that they are real 302 status code pages.

Reply:
today I learned!